packages/mask/src/extension/popups/pages/Wallet/ConnectWallet/index.tsx

```diff
@@ -81,6 +82,10 @@ const ConnectWalletPage = memo(() => {
             provider: Web3Helper.Web3ProviderDescriptor<NetworkPluginID.PLUGIN_EVM>,
         ) => {
             if (provider.type === ProviderType.MaskWallet) {
+                if (isLocked && !getLockStatusLoading) {
```

Comment:
No, Because the connect wallet page has other wallets (eg: metamask). So There should navigate the unlock page when the user selects the Mask wallet and adds some parameters to the route. About this logic you can refer to [jira](https://mask.atlassian.net/browse/MF-584)